### PR TITLE
Added an `ogbn-papers100M` example

### PR DESCRIPTION
**Pull request by puririshi98**
single gpu papers100m on 8xh100:   0.00027184496005873887 s/iter

### CODE REVIEW

File: examples/ogbn_papers_100m.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+if torch.cuda.is_available():
+    device = torch.device('cuda')
+else:
+    device = torch.device('cpu')
+
+dataset = PygNodePropPredDataset(name='ogbn-papers100M')
+split_idx = dataset.get_idx_split()
+data = dataset[0]
+data.y = data.y.reshape(-1).to(torch.long)
```
Comment (rusty1s) [inline examples/ogbn_papers_100m.py:33]: I am pretty certain this explains your index issues. Can you drop this here (since `NaN` cannot be converted to `long`), and just do the conversion inside each mini-batch?

File: examples/ogbn_papers_100m.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                             batch_size=batch_size, num_workers=pyg_num_work())
+test_loader = NeighborLoader(data, num_neighbors=[args.fan_out, args.fan_out],
+                             input_nodes=split_idx['test'],
+                             batch_size=batch_size, num_workers=pyg_num_work())
+eval_steps = 100
+acc = Accuracy(task="multiclass", num_classes=dataset.num_classes).to(device)
+for epoch in range(args.epochs):
+    for i, batch in enumerate(train_loader):
+        if i >= 10:
+            start = time.time()
```
Comment (akihironitta) [inline examples/ogbn_papers_100m.py:82]: nit: do we need this?
```suggestion
```

**Review by akihironitta — APPROVED** [review]

File: examples/ogbn_papers_100m.py (hunk 2 above)
Comment (puririshi98) [reply examples/ogbn_papers_100m.py:82]: yes, when timing we usually need to allow for some warmup iterations to get accurate average iteration times

Comment (akihironitta) [reply examples/ogbn_papers_100m.py:79]: I was thinking that because this is not a microbenchmark, it'd still be reasonable to measure the overall time incl. warmups, but I'm fine with either way :)
